File: pset6/mario/mario.py
from cs50 import get_int

def main():
    height = 0;
    # Prompt user for height of pyramid.
    while True:
        height = get_int("Height: ")
        if (height > 0 and height <= 8):
            break
    # Loop through every layer number, 1, 2, 3 up to `height`.
    for i in range(1, height + 1, 1):
       layer(i, height)

def layer(current, maximum):
    spaces = (maximum - current)
    # Print the layer, spaces, blocks, 2-gap, blocks
    print(" " * spaces + "#" * current + "  " + "#" * current)

main()

File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

string layer(int number, int max);

int main(void) {
    int height;
    
    // Query the user for their height.
    do {
        height = get_int("Height: ");
    } while (height < 1 || height > 8);
    
    // Generate the pyramid layer by layer
    for (int i = 0; i < height; i++) {
        layer(i + 1, height);
        printf("\n");
    }
}

// Pyramid Layer Function
string layer(int number, int max) {
    int spaces = max - number;
    // Leading whitespace
    for (int i = 0; i < spaces; i++) {
        printf("%s", " ");
    };
    // Bricks
    for (int i = 0; i < number; i++) {
        printf("%s", "#");
    };
    // Gap
    printf("%s", "  ");
    // Bricks
    for (int i = 0; i < number; i++) {
        printf("%s", "#");
    };
    return 0;
}